cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AuthorizationHandler.java

```diff
@@ -129,6 +148,91 @@ public void revoke(HttpRequest httpRequest, HttpResponder httpResponder) throws
     createLogEntry(httpRequest, request, HttpResponseStatus.OK);
   }
 
+  @Path("{principal-type}/{principal-name}/privileges")
+  @GET
+  public void listPrivileges(HttpRequest httpRequest, HttpResponder httpResponder,
+                             @PathParam("principal-type") String principalType,
+                             @PathParam("principal-name") String principalName) throws Exception {
+    ensureAuthorizationEnabled();
+    Principal principal = new Principal(principalName, Principal.PrincipalType.valueOf(principalType.toUpperCase()));
+    httpResponder.sendJson(HttpResponseStatus.OK, authorizer.listPrivileges(principal),
+                           new TypeToken<Set<Privilege>>() {
+                           }.getType(), GSON);
```

Comment:
nit: intellij quirks, but move to previous line? Also perhaps better to convert the TypeToken to a static final field so its not created during every request